packages/ddp-server/livedata_server.js

```diff
@@ -1601,12 +1567,57 @@ _.extend(Server.prototype, {
     // cursor observe callbacks have fired when your callback is invoked. (We
     // can change this if there's a real use case.)
     if (callback) {
-      callback(exception, result);
-      return undefined;
+      promise.then(
+        result => callback(undefined, result),
+        exception => callback(exception)
+      );
+    } else {
+      return promise.await();
     }
-    if (exception)
-      throw exception;
-    return result;
+  },
+
+  // @param options {Optional Object}
+  applyAsync: function (name, args, options) {
```

Comment:
Switched back to -Async, per discussion on Slack:
* -Async works better as an adverb than -Promise
* -Async is evocative of / aligned with ES7 `async` functions, which always return `Promise`s
* in a future where we have `async` functions, ideally one won't have to deal directly with things called `Promise`s